covariance/covarianceXsec.cpp

```diff
@@ -80,11 +85,18 @@ void covarianceXsec::InitXsecFromConfig() {
       throw;
     }
 
+	//ETA - we really should change this... good ol' b for flux...
     if(_fFancyNames[i].find("b_")==0) isFlux[i] = true;
     else isFlux[i] = false;
     i++;
   }
 
+  //Add a sanity check,
+  if(_fSplineNames.size() != SplineCounter){
+	MACH3LOG_ERROR("_fSplineNames is of size {} but found {} spline parameters", _fSplineNames.size(), SplineCounter);
+	throw;
```

Comment:
Can you use  throw MaCh3Exception(__FILE__ , __LINE__ ); instead of throw; ?

Reply:
yep done, changed this in some other places as well.